docs/workflow/debugging/coreclr/debugging-crossgen2.md

```diff
@@ -0,0 +1,65 @@
+
+- Since Crossgen2 is by default multi-threaded, it produces results fairly quickly even when compiling using a Debug variant of the JIT. In general, when debugging JIT issues we recommend using the debug JIT regardless of which environment caused a problem.
+
+- Crossgen2 supports nearly arbitrary cross-targetting, including OS and architecture cross targeting. The only restriction is that 32bit architecture cannot compile targetting a 64bit architecture. This allows the use of the debugging environment most convenient to the developer. In particular, if there is an issue which crosses the managed/native boundary, it is often convenient to debug using the mixed mode debugger on Windows X64.
+  - If the correct set of assemblies/command line arguments are passed to the compiler Crossgen2 should produce binary identical output on all platforms.
+  - The compiler does not check the OS/Architecture specified for input assemblies, which allows compiling using a non-architecture/OS matched version of the framework to target an arbitrary target. While this isn't useful for producing the diagnosing all issues, it can be cheaply used to identify the general behavior of a change on the full swath of supported architectures.
+
+Control compilation behavior by using the `--targetos` and `--targetarch` switches. The default behavior is to target the crossgen2's own OS/Arch pair, but all 64bit versions of crossgen2 are capable of targetting arbitrary OS/Arch combinations.
+At the time of writing the current supported sets of valid arguments are:
+| Command line arguments
+| --- |
+| `--targetos windows --targetarch x86` |
+| `--targetos windows --targetarch x64` |
+| `--targetos windows --targetarch arm` |
+|`--targetos windows --targetarch arm64` |
+|`--targetos linux --targetarch x64` |
+|`--targetos linux --targetarch arm` |
+|`--targetos linux --targetarch arm64` |
+|`--targetos osx --targetarch x64` |
+|`--targetos osx --targetarch arm64` |
+
+- Passing special jit behavior flags to the compiler is done via the `--codegenopt` switch. As an example to turn on tailcall loop optimizations and dump all code compiled use a pair of them like `--codegenopt NgenDump=* --codegenopt TailCallLoopOpt=1`.
+
+- When using the NgenDump feature of the JIT, disable parallelism as described above or specify a single method to be compiled. Otherwise, output from multiple functions will be interleaved and inscrutable.
+
+- Since there are 2 jits in the process, when debugging in the JIT, if the souce files match up, there is a decent chance that a native debugger will stop at unfortunate and unexpected locations. This is extremely annoying, and to combat this, we generally recommend making a point of using a runtime which doesn't exactly match that of the crossgen2 in use. However, if that isn't feasible, it is also possible to disable symbol loading in most native debuggers. For instance, in Visual Studio, one would use the "Specify excluded modules" feature.
```

Comment:
How does crossgen2 choose and find the JIT? How can you override which JIT is used?

Reply:
Crossgen2 will find the jit to use by looking in the same directory as the crossgen2.dll binary. It uses a naming convention to identify the exact jit dll to use.

In addition, there is support for a `--jitpath` switch to use a specific JIT. This option is intended to support A/B testing by the JIT team. The `--jitpath` option should only be used if the jit interface has not been changed. The JIT specified by the `--jitpath` switch must be compatible with the current settings of the `--targetos` and `--targetarch` switches.